### PR TITLE
Update changelog for 0.40 release

### PR DESCRIPTION
**Pull request by gbp**
## Relevant issue(s)

Required for #6347 

## What does this do?

Update changelog

### CODE REVIEW

File: doc/CHANGES.md [hunk 1]
```diff
@@ -3,28 +3,150 @@
...
 ## Upgrade Notes
 
+* This release will be the last to support Ruby 2.5 and 2.6. Please upgrade to
+  version 2.7.
 * Support for Ubuntu Bionic (18.04 LTS) will be removed in or after the next
   release.
+* Support for Debian 9 Stretch will be removed in or after the next release.
+* You might see `git protocol on port 9418 is no longer supported` when
+  deploying. Please switch to `https` for any theme URLs in your
+  `config/general.yml` See: #6630
```
Comment (gbp) [inline doc/CHANGES.md:55]:
```suggestion
  `config/general.yml` See: https://github.com/mysociety/alaveteli/pull/6630
```

File: doc/CHANGES.md [hunk 2]
```diff
@@ -3,28 +3,150 @@
 ## Highlighted Features
 
 * Use consistent colour for admin interface save buttons (Gareth Rees)
+* Remove dependency on Python (Graeme Porteous)
+* Use human-friendly URLs to body listings (Gareth Rees)
+* Add unsubscribe links to survey mailer (Graeme Porteous)
+* Prepare for the Rails 6.1 version (Graeme Porteous)
+* Add script to redact raw emails (Graeme Porteous)
```
Comment (garethrees) [inline doc/CHANGES.md:10]: `s/Add script/Add beta script/` just to signify that its not particularly battle-tested?

File: doc/CHANGES.md (hunk 1 above)
Comment (garethrees) [reply doc/CHANGES.md:55]: +1

File: doc/CHANGES.md [hunk 3]
```diff
@@ -3,28 +3,150 @@
...
+  version 2.7.
 * Support for Ubuntu Bionic (18.04 LTS) will be removed in or after the next
   release.
+* Support for Debian 9 Stretch will be removed in or after the next release.
+* You might see `git protocol on port 9418 is no longer supported` when
+  deploying. Please switch to `https` for any theme URLs in your
+  `config/general.yml` See: #6630
+* Controller actions now only respond to HTML requests. If your theme includes
+  custom actions which respond to JSON or other formats you will need to skip
+  before action callback.
```
Comment (garethrees) [inline doc/CHANGES.md:58]:
```diff
- skip before action callback.
- skip a `before_action` callback.
```

**Review by garethrees — APPROVED** [review]
Looks great!

Couple of minor nitpick suggestions but all good 😎

File: doc/CHANGES.md (hunk 3 above)
Comment (gbp) [reply doc/CHANGES.md:58]: Assuming you meant

```diff
+ skip a `before_action` callback.
```

Comment (garethrees) [reply doc/CHANGES.md]: oops, yeah!

**Pull request merged**